python/samples/demos/guided_conversations/guided_conversation/utils/conversation_helpers.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from dataclasses import dataclass, field
+import datetime
+from enum import Enum
+import logging
+from typing import Literal, Union
+
+
+class ConversationMessageType(Enum):
+    DEFAULT = "default"
+    ARTIFACT_UPDATE = "artifact-update"
+    REASONING = "reasoning"
+
+
+@dataclass
+class ConversationMessage:
```

Comment:
would be interesting to see if you could replace this with ChatMessageContent, with turn_number and timestamp in metadata, that way we also enable multi-modal and leverage all the code around that, can also be a matter of subclassing that with turn_number and timestamp as additional fields.

Reply:
Great suggestion, added! Resolved